src/OrchardCore/OrchardCore.Secrets.Abstractions/Models/RSASecret.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+
+namespace OrchardCore.Secrets.Models;
+
+public class RSASecret : SecretBase
+{
+    private byte[] _publicKeyAsBytes;
+    private byte[] _privateKeyAsBytes;
+
+    public string PublicKey { get; set; }
+    public string PrivateKey { get; set; }
+    public RSAKeyType KeyType { get; set; }
+
+    public byte[] PublicKeyAsBytes()
+    {
+        if (_publicKeyAsBytes is not null)
+        {
+            return _publicKeyAsBytes;
+        }
+
+        return _publicKeyAsBytes = KeyAsBytes(PublicKey);
+    }
+
+    public byte[] PrivateKeyAsBytes()
```

Comment:
I dunno if "secret references" - i.e secrets that don't embed the crypto material - is a scenario you want to support, but if we decide to provide such an API, we won't be able to support cases where the private key is stored in a HSM and can't be retrieved by the application itself. Just exposing a method that returns a `RSA` instance that is used to perform the crypto operations might be a better approach.

Reply:
For now we have 2 `TextSecret` and `RSASecret` types derived from a `SecretBase` type, `RSASecret` as defined here always hold the keys that are populated from different `ISecretStore`: database, configuration, keyvault. 

Here I did this method just to cache the result of `KeyAsBytes()` that is needed in different places.

Can the specific case of HSM be handled by a specific store? Maybe Azure Key Vault?